src/VisualStudio/Core/SolutionExplorerShim/HierarchyItemToProjectIdMap.cs

```diff
@@ -47,6 +46,16 @@ public bool TryGetProjectId(IVsHierarchyItem hierarchyItem, string targetFramewo
             var project = _workspace.DeferredState.ProjectTracker.ImmutableProjects
                     .Where(p =>
                     {
+                        // We're about to access various properties of the IVsHierarchy associated with the project.
```

Comment:
This is needed for matching nodes in Solution Explorer to their corresponding Roslyn project. This is used to determine if we should show the Analyzers folder under the References node, and also in associating nodes for analyzers to the proper `AnalyzerReference`.